cmd/podman/machine/init.go

```diff
@@ -148,6 +149,9 @@ func initMachine(cmd *cobra.Command, args []string) error {
 
 	// Process optional flags (flags where unspecified / nil has meaning )
 	if cmd.Flags().Changed("user-mode-networking") {
+		if runtime.GOOS != "windows" {
+			return fmt.Errorf("setting --user-mode-networking is not supported on %q platforms", runtime.GOOS)
+		}
```

Comment:
What is the rationale behind this? This should not hardcode the OS IMO, this a backend specific option. 
Also qemu, hyperv and apple hypervisor all use gvproxy so they are already using user mode networking.
There is a warning in the qemu backend when it is set to false: https://github.com/containers/podman/blob/93f194cfb5401778c4f2e848ae5d7552ad182dcd/pkg/machine/qemu/machine.go#L376-L378

Also given that the option is now already accepted it means that this would be a breaking change so I argue we should not do this. cc @n1hility

Reply:
The way I documented it was: "When the qemu backend is used (Linux, Mac), user-mode networking is mandatory and the only allowed value is true. In contrast, The Windows/WSL backend defaults to false, and follows the standard WSL network setup. Changing this setting to true on Windows/WSL informs Podman to replace the WSL networking setup on start of this machine instance with a user-mode networking distribution <snip>"

The way I was thinking of it was that we may in the future want (or receive a contribution for) VM built-in virt net NAT. For example, for performance benefits. So in the future a value of false may be allowed / make sense.